packages/cojson/src/localNode.ts

```diff
@@ -445,17 +456,15 @@ export class LocalNode {
       return;
     }
 
-    const groupAsInvite = expectGroup(
-      group.core
-        .testWithDifferentAccount(
-          new ControlledAgent(inviteAgentSecret, this.crypto),
-          this.crypto.newRandomSessionID(inviteAgentID),
-        )
-        .getCurrentContent(),
+    const { oldAgentSecret, oldSessionID } = this.switchControl(
+      inviteAgentSecret,
+      this.crypto.newRandomSessionID(inviteAgentID),
     );
```

Comment:
Calling switchControl in a fully loaded node is definitely too unpredictable and disruptive for the higher level of abstraction (mostly jazz-tools)

I think the previous version of spawning a LocalNode is a little easier to manage than a temporary ownership transfer.